app/ai_backend/test_genesis_connector.py

```
@@ -789,56 +1674,101 @@ def test_send_request_with_binary_payload(self, mock_post):
         mock_response.json.return_value = {'data': 'nested_processed'}
         mock_post.return_value = mock_response
         
-        nested_payload = {
-            'level1': {
-                'level2': {
-                    'level3': {
-                        'level4': {
-                            'message': 'deeply nested'
-                        }
-                    }
+        for timeout in boundary_timeouts:
```

Ruff (0.11.9)

Undefined name `boundary_timeouts`

(F821)